examples/custom-node/src/primitives/extended_op_tx_envelope.rs

```diff
@@ -173,20 +171,21 @@ where
     }
 }
 
-impl<T> Decodable2718 for ExtendedOpTxEnvelope<T>
+impl<B, T> Decodable2718 for ExtendedTxEnvelope<B, T>
 where
-    T: Decodable2718,
+    B: Decodable2718 + SerdeBincodeCompat,
+    T: Decodable2718 + SerdeBincodeCompat,
 {
     fn typed_decode(ty: u8, buf: &mut &[u8]) -> Eip2718Result<Self> {
         match ty.try_into() {
             Ok(tx_type) => match tx_type {
                 OpTxType::Eip2930 | OpTxType::Eip1559 | OpTxType::Eip7702 | OpTxType::Deposit => {
-                    let envelope = OpTxEnvelope::typed_decode(ty, buf)?;
-                    Ok(Self::BuiltIn(envelope))
+                    let envelope = B::typed_decode(ty, buf)?;
+                    Ok(Self::BuiltIn(B::from_repr(envelope.as_repr())))
                 }
                 OpTxType::Legacy => {
-                    let envelope = OpTxEnvelope::typed_decode(ty, buf)?;
-                    Ok(Self::BuiltIn(envelope))
+                    let envelope = B::typed_decode(ty, buf)?;
+                    Ok(Self::BuiltIn(B::from_repr(envelope.as_repr())))
                 }
```

Comment:
for this we need a different workaround but we can make this work in another followup